### PR TITLE
Allow . in c_names and cxx_names

### LINKED ISSUE #2033: dune does not find C files with a dot in the name
With dune 1.8.2.
Repro:
- an empty hello_world.ml file
- an empty C file with a '.' dot in the name, like `file.a.c`
- a dune file:
```
(library
 (name hello_world)
 (c_names file.a)
)
```
Then with `dune build` one gets the misleading error:
```
File "dune", line 3, characters 10-16:
3 |  (c_names file.a)
              ^^^^^^
Error: file.a does not exist as a C source. file.a.c must be present
```
Changing the name to `file_a.c` for example works.

### PR DESCRIPTION
**Pull request by rgrinberg**
Let's get a confirmation fro m @diml that this should be allowed. The fix itself is trivial.

Fix #2033

### CODE REVIEW

**Review by aalekseyev — APPROVED** [review]
Any reason the test is negative (showing the error message) rather than positive (showing a successful compilation)?

Assuming the compilation succeeds when the file is added, this change seems obviously good to me.

**Comment by rgrinberg** [comment]
The reason is that I forgot to promote. Thanks for noticing.

**Comment by rgrinberg** [comment]
Sorry @ejgallego I must have added you as a reviewer accidentally

File: test/blackbox-tests/test-cases/github2033/run.t [hunk 1]
```diff
@@ -0,0 +1,5 @@
+. should be allowed in c names
+  $ dune build
+  Info: creating file dune-project with this contents:
+  | (lang dune 1.9)
```
Comment (emillon) [inline test/blackbox-tests/test-cases/github2033/run.t]: the project file is missing in the test :)

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/github2033/run.t]: Thanks, I added one.

**Comment by diml** [comment]
I can't think of a reason why this shouldn't be allowed. I can't remember either why it wasn't allowed in the first place.

**Pull request merged**